src/main/java/net/gnomeffinway/depenizen/extensions/protocolsupport/ProtocolSupportExtension.java

```diff
@@ -0,0 +1,62 @@
+package net.gnomeffinway.depenizen.extensions.protocolsupport;
+
+import net.aufdemrand.denizencore.objects.Element;
+import net.aufdemrand.denizencore.objects.dObject;
+import net.aufdemrand.denizen.objects.dPlayer;
+import net.aufdemrand.denizencore.tags.Attribute;
+import net.gnomeffinway.depenizen.extensions.dObjectExtension;
+import org.bukkit.entity.Player;
+import protocolsupport.api.ProtocolSupportAPI;
+
+public class ProtocolSupportExtension extends dObjectExtension {
+
+    public static boolean describes(dObject pl) {
+        return pl instanceof dPlayer;
+    }
+
+    public static ProtocolSupportExtension getFrom(dObject pl) {
+        if (!describes(pl)) return null;
+        else return new ProtocolSupportExtension((dPlayer) pl);
+    }
+
+    public ProtocolSupportExtension(dPlayer player) {
+        this.player = player.getPlayerEntity();
+    }
+
+    Player player = null;
+
+    @Override
+    public String getAttribute(Attribute attribute) {
+        // <--[tag]
+        // @attribute <p@player.version>
+        // @returns Element
+        // @description
+        // Returns the version of player.
+        // @plugin Depenizen, ProtocolSupport
+        // -->
+        if (attribute.startsWith("version")) {
+            if(player == null)
+                return null;
+
+            attribute = attribute.fulfill(1);
+
+            // <--[tag]
+            // @attribute <p@player.version>
```

Comment:
I didn't notice that.
I don't know, but <p@player.version> will return the client version based on protocol version, example. MINECRAFT_1_8
and <p@player.version.protocol> will return the protocol version, like 47 for 1.8.x client.